pepys_timeline/static/js/src/pepys.js

```diff
@@ -179,49 +300,54 @@ function renderCharts(serials) {
     }
 }
 
+function renderTimelines() {
+    const transformedSerials = transformSerials2();
+    console.log('transformedSerials2', transformedSerials);
 
+    if (!generatedCharts) {
+        console.log('Generating charts.');
+        for (i = 0; i < transformedSerials.length; i++) {
+            console.log(transformedSerials[i].name, transformedSerials[i].overall_average);
 
-window.onload = (event) => {
-  fetch('/config')
-    .then(response => response.json())
-    .then(response => {
-        const { frequency_secs } = response;
-        fetchSerials();
-        setInterval(fetchSerials, frequency_secs * 100);
+            if (!transformedSerials[i].includeInTimeline) {
+                console.log("Serial flag 'includeInTimeline' false, won't generate chart.");
+                continue;
+            }
 
-    })
-    .catch(err => console.error(err));
-
-  const serial_participants = [{
-    "serial_id": "faa18ef7-6823-33dc-0f16-de6e4b2c02f3",
-    "platform_id": "50d64387-9f91-4c6f-8933-f4e7b1a7d8ab",
-    "start": "2020-11-15 00:00:00",
-    "end": "2020-11-15 23:59:59",
-    "gap_seconds": 30
-  }];
-  const range_types = ["G", "C"];
+            // take deep copy of data. For some reason using a dataset
```

Comment:
I'm pretty sure we don't need to do this deep copy any more @krsnik93 , we're not re-using mock datasets any more.